File: 2_String/Basic/657_Robot_Return_to_Origin.cpp
""" Amazon
Basic simulation
"""
class Solution {
public:
    bool judgeCircle(string moves) {
        vector<int> coor{0, 0};
        for(char m: moves){
            if(m=='U'){
                coor[0] += 1;
            }
            if(m=='D'){
                coor[0] -= 1;
            }
            if(m=='L'){
                coor[1] += 1;
            }
            if(m=='R'){
                coor[1] -= 1;
            }
        }
        return coor[0]==0 && coor[1]==0;
    }
};

File: A_Basic/String/657_Robot_Return_to_Origin.py
""" Amazon
Basic simulation
"""


class Solution:
    def judgeCircle(self, moves: str) -> bool:
        coor = [0, 0]
        for m in moves:
            if m == 'U':
                coor[0] += 1
            if m == 'D':
                coor[0] -= 1
            if m == 'L':
                coor[1] += 1
            if m == 'R':
                coor[1] -= 1
        if coor == [0, 0]:
            return True
        else:
            return False
